### PR TITLE
[Admin] Delete subcomments

### PR DESCRIPTION
**Pull request by ghost**
## Description
Delete subcomment
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- routes/web.php

## SQL query and `explain`

`explain select * from `comments` where `comments`.`id` = 45  limit 1\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: comments
>    partitions: NULL
>          type: const
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: NULL
> 1 row in set, 1 warning (0.00 sec)

`explain update `comments` set `deleted_at` = "2018-05-21 06:27:17", `updated_at` ="2018-05-21 06:27:17"  where `id` = 45\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: UPDATE
>         table: comments
>    partitions: NULL
>          type: range
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: Using where
> 1 row in set (0.00 sec)

### CODE REVIEW

**Comment by at-phuongvu** [comment]
```
        modified:   app/Http/Controllers/Admin/PostController.php
        modified:   public/js/messages.js
        modified:   public/js/post.js
        modified:   resources/lang/en/post.php
        modified:   resources/views/admin/pages/posts/show.blade.php
        modified:   routes/web.php
        modified:   app/Models/Comment.php
```

File: routes/web.php [hunk 1]
```diff
@@ -21,6 +21,7 @@
     Route::resource('categories', 'CategoryController')->parameters(['categories' => 'id']);
     Route::resource('products', 'ProductController')->parameters(['products' => 'id']);
     Route::resource('posts', 'PostController')->parameters(['posts' => 'id']);
+    Route::delete('posts/{post_id}/{id}', 'PostController@deleteComment')->name('posts.comment.destroy');
```
Comment (at-xinhnguyen) [inline routes/web.php]: `(post/{post_id}/comments/{id})` hoặc `(comments/{id})` đều đc.
- chỗ này em dùng `CommentController` nhé.

Comment (at-phuongvu) [reply routes/web.php]: `Comment` tạo ra comment controller cho 1 cái method đó thôi hả anh ...

File: resources/views/admin/pages/posts/show.blade.php [hunk 2]
```diff
@@ -51,8 +54,10 @@
                       <form action="" class="col-md-4">
                         <button class="btn btn-primary" type="submit"><i class="fa fa-edit icon-size" ></i></button>
                       </form>
-                      <form action="" class="col-md-4">
-                        <button class="btn btn-primary" type="submit"><i class="fa fa-trash icon-size" ></i></button>
+                      <form action="{{ route('admin.comments.destroy', ['id' => $comment['id']]) }}" class="col-md-4" method="POST" id="delete{{ $comment['id'] }}">
+                        @csrf
+                        @method('DELETE')
+                        <button onclick="deleteComment(event, {{ $comment['id'] }})" class="btn btn-danger" type="submit"><i class="fa fa-trash icon-size" ></i></button>
```
Comment (at-xinhnguyen) [inline resources/views/admin/pages/posts/show.blade.php]: giao diện này a nghĩ dùng ajax xử lý đỡ load trang có vẽ hay hơn ấy.

File: public/js/post.js [hunk 3]
```diff
@@ -40,5 +40,33 @@ function loadAjax(url, id) {
     csrf = document.getElementById('csrf-token').getAttribute('content');
     xmlhttp.setRequestHeader('X-CSRF-Token', csrf);
 
+    xmlhttp.send();
+}
+
+function deleteComment(e, id, url) {
+    e.preventDefault();
+    msg = Lang.get('post.admin.form.delete_comment_msg') + id;
+    console.log(url);
```
Comment (at-xinhnguyen) [inline public/js/post.js]: remove line này đi em

File: routes/web.php [hunk 4]
```diff
@@ -21,6 +21,7 @@
     Route::resource('categories', 'CategoryController')->parameters(['categories' => 'id']);
     Route::resource('products', 'ProductController')->parameters(['products' => 'id']);
     Route::resource('posts', 'PostController')->parameters(['posts' => 'id']);
+    Route::resource('comments', 'CommentController')->parameters(['comments' => 'id']);
```
Comment (at-xinhnguyen) [inline routes/web.php]: ajax thì nên để trong bên api em nhé

File: app/Http/Controllers/Admin/CommentController.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\Comment;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class CommentController extends Controller
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Admin/CommentController.php]: class này k dùng thì xoá đi em

**Review by ghost — CHANGES_REQUESTED** [review]
Please check my cmt

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**